src/status_im/common/theme/events.cljs

```diff
@@ -2,23 +2,36 @@
   (:require
     [re-frame.core :as re-frame]
     [status-im.common.theme.core :as theme]
-    [status-im.common.theme.utils :as utils]))
+    [status-im.constants :as constants]))
 
 (re-frame/reg-fx
  :theme/init-theme
  (fn []
    (theme/add-device-theme-change-listener)))
 
+(defn- appearance-type->theme
+  "Converts appearance type identifier to a theme keyword.
+  Returns `:light` or `:dark` based on `appearance-type`:
+  - `appearance-type-light` (1): Light theme.
+  - `appearance-type-dark` (2): Dark theme.
+  - `appearance-type-system` (0): Uses system preference."
+  [appearance-type]
+  (condp = appearance-type
+    constants/appearance-type-dark   :dark
+    constants/appearance-type-light  :light
+    constants/appearance-type-system (if (theme/device-theme-dark?) :dark :light)
+    :dark))
+
 (re-frame/reg-event-fx
  :theme/switch
- (fn [{db :db} [{:keys [theme view-id theme-type]}]]
+ (fn [{db :db} [{:keys [theme view-id appearance-type]}]]
    "Switches the theme and triggers related effects.
    Parameters:
-   - theme: Optional theme keyword to apply. If not provided, defaults to the theme determined by theme-type.
-   - theme-type: Optional theme type to determine the theme. If not provided, defaults to the theme type from the user's profile.
-   - view-id: Optional view ID for updating status color. If not provided, uses the current view ID from the database."
-   (let [theme-type (or theme-type (get-in db [:profile/profile :appearance]))
-         theme      (or theme (utils/theme-type->theme-value theme-type))]
+   - theme: Optional theme keyword to apply. If not provided, defaults to the theme determined by appearance-type.
+   - appearance-type: Optional appearance type to determine the theme. If not provided, defaults to the theme type from the user's profile.
+   - view-id: Optional view ID for updating status and nav color. If not provided, uses the current view ID from the database. View-id required because status and nav bar colors depend on the screen. For example, on the home screen, the navigation color should always be dark, regardless of the theme, due to the bottom tabs"
```

Comment:
@Parveshdhull, a minor problem from the original code that's being carried over to this PR. The apparent "docstring" is actually a normal string being allocated on every dispatch to `:theme/switch`. In re-frame apps I've seen the more common approach of describing the event behavior as a comment right before the call to `reg-event-fx`. This will also allow zprint to format the comment with the correct number of chars per line.

Reply:
Thank you, updated